actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -84,7 +84,45 @@ def formats
         end
       end
 
-      # Sets the \variant for template.
+      # Sets the \variant for the response template.
+      #
+      # Action View will incorporate the request's variants when determining which template to render. For example, if an <tt>ArticlesController</tt> needs to respond to an <tt>#index</tt> action request with <tt>request.variant = [:ios, :turbo_native]</tt>, it will render the first template file it can find in the following list:
+      #
+      # - `app/views/articles/index.html+ios.erb`
+      # - `app/views/articles/index.html+turbo_native.erb`
+      # - `app/views/articles/index.html.erb`
+      #
+      # Variant names can refer to any aspect of a request, from the type of
+      # User Agent (e.g. iOS, Android, etc.), to the version of the requesting
+      # application (e.g. v1, v2, etc.) to the level of permissions available to
+      # the requesting User (e.g. administrator, contributor, guest, etc.).
+      #
+      # A word of caution: be stingy when adding new variants. Only reach for
+      # variants when groups of conditionals (that share a theme) or other
+      # more traditional templating techniques become unmanageable.
```

Comment:
Similar to https://github.com/rails/rails/pull/50273#discussion_r1416160206, I'm interested in workshopping the guidance provided here.

Reply:
I would just `s/stingy/cautious` since "stingy" is not really a basic term people are familiar with.

Reply:
I re-worded the block and swapped `stingy` with `judicious`. Is that appropriate?

Reply:
I'm not sure about `judicious` either, and it doesn't appear in [this dictionary](https://jisho.org/search/judicious) that I'm often using. Not that should be a requirement for the lexicon used in the docs, but it is a telling sign.

Personally, I prefer to use simpler terms that are used on a regular basis, "judicious" is not something I would use with my ESL wife or children. 😂 

In this case, I might write it like:

> Note: When adding many new variant templates with similarities to existing template files can make maintaining your view code more difficult.

Here we are cutting to the chase, "having many variants can make unmaintainable code" is the essential thing we want to say here so try to do that in as few words as possible.

I still think it's good to explain "why" they become unmaintainable at some point.

There is a similar statement below in the `layouts_and_rendering` guide that should also be addressed.

Reply:
Thank you!